src/ray/raylet/node_manager.cc

```diff
@@ -3007,6 +3007,16 @@ MemoryUsageRefreshCallback NodeManager::CreateMemoryUsageRefreshCallback() {
             ray::stats::STATS_memory_manager_worker_eviction_total.Record(
                 1, "MemoryManager.ActorEviction.Total");
           }
+        } else {
+          const static int64_t max_to_print = 10;
+          auto all_workers = this->WorkersWithLatestSubmittedTasks(
```

Comment:
I thought if there are no retryable tasks, we should then try to kill one that isn't retry-able, and prioritize tasks over actors? Otherwise, won't it be a regression for cases where the application can't finish?

I think the ideal case would be to kill a non-retryable task or actor, and then customize the error message to explain that if they want it to be retried, the user should specify max_restarts or max_retries. Otherwise, we'll only have the generic WorkerDiedError.

Reply:
i wonder how likely this will happen in practical though. The trade off is when this case happens, 
- either we fail the job by application level killer;
- or the job fails anyway because system oom killer kicks in (but with less ideal error message, which we can improve here?);
- or the job passes;

Reply:
I think the policy we want is already happening with default settings which I assume is the majority case (3 task retry , 0 actor restart), where tasks are killed before actor

So I think the change here is to kill non-retriable task and provide a message indicating that is happening, and caller should consider making things retriable